tests/test_asset.py

```diff
@@ -306,3 +312,149 @@ def test_failure_bogus_uri():
     """Raise ValueError."""
     with pytest.raises(ValueError):
         asset.info("bogus")
+
+
+@pytest.fixture(scope="session")
+def user_workspace():
+    """A test user with its own workspace."""
+    # Ensure that user and workspace are unique to the session.
+    TAG = str(uuid.uuid4())[:8]
+    DISPLAY_NAME = "Test User {TAG}"
+    EMAIL = f"testuser.{TAG}@example.com"  # dash not allowed in email?
+    USERNAME = f"testuser-{TAG}"
+    PASSWORD = "password"
+    WORKSPACE = f"workspace-{TAG}"
+
+    # Four different secrets are required in the environment.
+    MYSQL_PASSWORD = os.getenv("MYSQL_PASSWORD")
+    AWS_ACCESS_KEY_ID = os.getenv("AWS_ACCESS_KEY_ID")
+    AWS_SECRET_ACCESS_KEY = os.getenv("AWS_SECRET_ACCESS_KEY")
+    AWS_S3_PATH = os.getenv("AWS_S3_PATH")
```

Comment:
Dummy values for the last three secrets will suffice. The test does not require cloud storage.